modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -702,12 +688,12 @@ FeatureFloodCount::flood(const DofObject * dof_object, int current_idx, FeatureD
    * If we reach this point (i.e. we haven't returned early from this routine),
    * we've found a new mesh entity that's part of a feature.
    */
-  unsigned int map_num = _single_map_mode ? 0 : current_idx;
-  processor_id_type rank = processor_id();
+  auto map_num = _single_map_mode ? decltype(current_idx)(0) : current_idx;
+  auto rank = processor_id();
 
   // New Feature (we need to create it and add it to our data structure)
   if (!feature)
-    _partial_feature_sets[rank][map_num].push_back(FeatureData(current_idx));
+    _partial_feature_sets[rank][map_num].emplace_back(FeatureData(current_idx));
```

Comment:
This isn't really the right way to use `emplace_back()`.  It will compile and work... but I won't give you the "perfect forwarding in-place construction".  What it's actually doing is creating a `FeatureData` and then (since it's an rvalue) it will be rvalue assigned (moved) into the memory spot (or, if there is no move operator for this object then it will be _copied_ into place... just like you would get with `push_back()`!!)

With `emplace_back()` you use "variadic templates" to actually just pass through the constructor arguments.  So this line of code should really look like:

```C++
_partial_feature_sets[rank][map_num].emplace_back(current_idx);
```

That `current_idx` will get passed through to the constructor and the object will actually just get created _once_ right in the right memory spot.

A _really_ good resource for reading about `emplace_back()` is here: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2007/n2345.pdf

(seriously - it's a good read... not just a technical document).